homework.3/homework.3.playground/Contents.swift

```diff
@@ -0,0 +1,121 @@
+// homework.3
+
+import Darwin
+
+numberAddition(10, 12)
+numberAddition(99.8, 100.2)
+
+numberSubtraction(100, 50)
+numberSubtraction(200.5, 100)
+
+numberDivision(20, 4)
+numberDivision(12.8, 2.0)
+
+numberMultiplication(2, 8)
+numberMultiplication(20.0, 5.1)
+
+sumDigits(1234)
+
+stringComparison("Hello", "!")
+
+exponentiation(2, 2)
+
+print("Factorial = \(factorial(-1))")
+
+// Addition Functions
+
+func numberAddition(_ firstNum: Int, _ secondNum: Int) {
+    print("intNumberAddition = \(firstNum + secondNum)\n")
+}
+
+func numberAddition(_ firstNum: Double, _ secondNum: Double) {
+    print("doubleNumberAddition = \(firstNum + secondNum)\n")
+}
+
+// Subtraction Functions
+
+func numberSubtraction(_ firstNum: Int, _ secondNum: Int) {
+    print("intNumberSubtraction = \(firstNum - secondNum)\n")
+}
+
+func numberSubtraction(_ firstNum: Double, _ secondNum: Double) {
+    print("doubleNumberSubtraction = \(firstNum - secondNum)\n")
+}
+
+// Multiplication Functions
+
+func numberMultiplication(_ firstNum: Int, _ secondNum: Int) {
+    print("intNumberMultiplication = \(firstNum * secondNum)\n")
+}
+
+func numberMultiplication(_ firstNum: Double, _ secondNum: Double) {
+    print("doubleNumberMultiplication = \(firstNum * secondNum)\n")
+}
+
+// Division Functions
+
+func numberDivision(_ firstNum: Int, _ secondNum: Int) {
+    print("intNumberDivision = \(firstNum / secondNum)\n")
+}
+
+func numberDivision(_ firstNum: Double, _ secondNum: Double) {
+    print("doubleNumberDivision = \(firstNum / secondNum)\n")
+}
```

Comment:
It's better to use verbs in the function namings when we can. For example, `addNumbers` instead of `numberAddition` and so on

It's also better to use full words not shortened in names. So `firstNumber`, not `firstNum`